packages/compass-components/src/components/confirmation-modal.tsx

```diff
@@ -1,17 +1,20 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import LeafyGreenConfirmationModal from '@leafygreen-ui/confirmation-modal';
 import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 const { track } = createLoggerAndTelemetry('COMPASS-UI');
 
-function ConfirmationModal(
-  props: React.ComponentProps<typeof LeafyGreenConfirmationModal> & {
-    trackingId?: string;
-  }
-): React.ReactElement {
-  if (props.open && props.trackingId) {
-    track('Screen', { name: props.trackingId });
-  }
-  return <LeafyGreenConfirmationModal {...props} />;
+function ConfirmationModal({
+  trackingId,
+  ...props
+}: React.ComponentProps<typeof LeafyGreenConfirmationModal> & {
+  trackingId?: string;
+}): React.ReactElement {
+  useEffect(() => {
+    if (props.open && trackingId) {
+      track('Screen', { name: trackingId });
+    }
+  }, [props.open, trackingId]);
+  return <LeafyGreenConfirmationModal data-testid={trackingId} {...props} />;
```

Comment:
Small drive-by change to stop leaking trackingid to the actual dom, shouldn't change behavior of the analytics at all